app/lib/submission_builder/ty2024/states/md/documents/md502.rb

```diff
@@ -10,13 +8,18 @@ class Md502 < SubmissionBuilder::Document
 
             def document
               build_xml_doc("Form502", documentId: "Form502") do |xml|
-                if @submission.data_source.direct_file_data.claimed_as_dependent?
+                xml.MarylandSubdivisionCode intake.subdivision_code
+                unless intake.political_subdivision&.end_with?("- unincorporated")
+                  xml.CityTownOrTaxingArea intake.political_subdivision
+                end
+                xml.MarylandCounty county_abbreviation
+                if intake.direct_file_data.claimed_as_dependent?
                   xml.FilingStatus do
                     xml.DependentTaxpayer "X"
                   end
-                elsif @submission.data_source.filing_status == :married_filing_separately
+                elsif intake.filing_status == :married_filing_separately
                   xml.FilingStatus do
-                    xml.MarriedFilingSeparately "X", spouseSSN: @submission.data_source.direct_file_data.spouse_ssn
+                    xml.MarriedFilingSeparately "X", spouseSSN: intake.direct_file_data.spouse_ssn
                   end
```

Comment:
we have a method on this xml builder for @submission.data_source to be called intake so thought to use it